File: Level_1/problem_1117.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Problem_1117 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<String> resultados = new ArrayList<>();
        List<Double> notas_validas = new ArrayList<>();

        while (notas_validas.size() < 2) {
            double n = scanner.nextDouble();
            if (n < 0 || n > 10) {
                resultados.add("nota invalida");
            } else {
                notas_validas.add(n);
            }
        }

        double media = (notas_validas.get(0) + notas_validas.get(1)) / 2;
        resultados.add(String.format("media = %.2f", media));

        for (String resultado : resultados) {
            System.out.println(resultado);
        }
        scanner.close();
    }
}

File: Level_1/problem_1117.py
resultados = []
notas_validas = []

while len(notas_validas) < 2:
    n = float(input())
    if n < 0 or n > 10:
        resultados.append("nota invalida")
    else:
        notas_validas.append(n)

media = sum(notas_validas) / 2
resultados.append(f"media = {media:.2f}")

for resultado in resultados:
    print(resultado)
